### PR TITLE
Add calendar tests

### PR DESCRIPTION
**Pull request by anamariamilcu**
Created 5 tests that will analyse the correctness of the next feature that will be added, where optional attendees will be taken into account. The tests explore multiple potential situations.

### CODE REVIEW

File: walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java [hunk 1]
```diff
@@ -121,6 +123,69 @@ public void everyAttendeeIsConsidered() {
     Assert.assertEquals(expected, actual);
   }
 
+  @Test
+  public void everyAttendeeIsConsideredAndOptionalIgnored() {
```
Comment (alexcepoi) [inline walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java:127]: Here and elsewhere: I didn't really understand what this test was testing just by looking at the name. It's generally very good practices to have test names be really clear on what behavior they are testing.

"considered" is too ambiguous here to make it clear what exactly the test expects.

Test names should be in third person and describe the behavior and circumstances if applicable.

E.g. "DoesXWhenY", not "TakesXIntoConsideration"

Good read: https://drive.google.com/file/d/0B_-S6UdTLV5pNnhTRnFzYnRHenM/view

Something like "PrefersTimeSlotWhereOptionalAttendeesCanAttend" "IgnoresOptionalAttendeesIfNoSuitableTimeSlotFound" might be clearer.

Sadly even the existing tests in this file are not following best practices.

File: walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java [hunk 2]
```diff
@@ -121,6 +123,69 @@ public void everyAttendeeIsConsidered() {
...
+  }
+
+  @Test
+  public void everyAttendeeIsConsideredAndOptionalIncluded() {
+    // Have each person have different events and an optional attendee who has a meeting exactly
+    // when the mandatory attendees have a free spot. This spot will be eliminated and return
+    // only the beggining of the day and the end o f the day.
+    //
+    // Events  :       |--A--|     |--B--|
+    // Optional:             |--C--|
```
Comment (alexcepoi) [inline walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java:165]: Optional: An interesting additional corner case here would be what would happen if between A and B's events, optional attendee C has an event, but there is still room for to schedule a meeting.

E.g. Are the intervals that work for A, B modified to match C's availability (better and I think what the code does) or just removed if they don't match C's availability.

Consider also some overlap between required and optional attendees.

File: walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java [hunk 3]
```diff
@@ -121,6 +123,69 @@ public void everyAttendeeIsConsidered() {
...
   }
 
+  @Test
+  public void everyAttendeeIsConsideredAndOptionalIgnored() {
+    // Have each person have different events and an optional attendee who has a meeting the
+    // whole day. Since he doesn't have any available spots, he will be ignored and 3 ranges
+    // will be returned.
+    //
+    // Events  :       |--A--|     |--B--|
+    // Optional: |------------C----------------|
```
Comment (alexcepoi) [inline walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java:133]: I don't think you have any test which considers what would happen if optional attendees:
- have partial availability: there is no slot that works for an optional attendee, but they could attend half of the required duration for some slots. Will those slots be preferred in any way?
- if you have 2 optional attendees and 3 possible slots: 1 only works for required attendees, 1 works for first optional ateendee, 1 works for second optional attendee. Which ones would the function return?

**Review by alexcepoi — APPROVED** [review]
Nice!
